driver/src/orderbook/streamed/updating_orderbook.rs

```diff
@@ -89,51 +83,71 @@ impl StableXOrderBookReading for UpdatingOrderbook {
 /// Returns Ok when exit_indicator is dropped.
 /// Returns Err if the stream ends.
 async fn update_with_events_forever(
+    web3: Web3,
+    contract: Arc<dyn StableXContract>,
     orderbook: Arc<Mutex<Orderbook>>,
     orderbook_ready: Arc<AtomicBool>,
     mut block_timestamp_reader: impl BlockTimestampReading,
     exit_indicator: oneshot::Receiver<()>,
-    past_events: impl Future<Output = Result<Vec<Event<batch_exchange::Event>>, ExecutionError>>,
-    stream: impl Stream<Item = Result<Event<batch_exchange::Event>, ExecutionError>>,
 ) -> Result<()> {
-    // `select!` requires the futures to be fused...
+    const POLL_INTERVALL: Duration = Duration::from_secs(30);
```

Comment:
If we want to poll, we probably want to do this much more frequently (and something we can configure via an ENV variable). However, this made me think that we might just want to fetch the last "gap" whenever the orderbook is queried and not have a thread at all (since the atomic var is preventing us from accessing the orderbook before we processed everything anyways).

Reply:
I think that might work nicely for the driver.

Reply:
I like this idea because it get rid of the thread.

Reply:
I want to implement that in a separate PR.

Reply:
How do you think the initial update should work? Still in the background or should there be a blocking init method? I don't think  it should happen on first orderbook request because it would block for too long.

Reply:
Maybe it could happen at startup time before the scheduler gets started. This way once we start the scheduler and solving, the orderbook is ready.

The downside is that we may miss some batches when we restart, but that happens anyway now.

Reply:
At the moment we are seeing a lot of "Orderbook not yet ready for fetching" errors on startup, because we are blocking access to the orderbook until the first fetch is complete. 

I agree that we should not waste the initial waiting time (if we start between minute 3 and minute 5) and use it to fetch the orderbook, but I think it can happen on a blocking thread, since we cannot really do any processing before anyways.